docs/_blog/first_bdk_taproot_tx.md

```diff
@@ -0,0 +1,665 @@
+---
+title: "The first BDK Taproot TX: a look at the code (Part 1)"
+description: "A quick overview of the changes made to bdk, rust-miniscript and rust-bitcoin to make a Taproot transaction"
+authors:
+    - Alekos Filini
+date: "2021-11-15"
+tags: ["BDK", "taproot", "miniscript"]
+permalink: "/blog/2021/11/first-bdk-taproot-tx-look-at-the-code-part-1"
+---
+
+This is the first of a two-parts blog series in which I will try to explain all the changes that I made to BDK (and some of its dependencies) to make our [first Taproot transaction in mainnet][first-mainnet-tx], which also
+turned out to be [the first ever use of the new `OP_CHECKSIGADD` opcode][first-ever-use-checksigadd].
+
+Hopefully this will give an insight into what kind of changes need to be made to a wallet in order to support spending `P2TR` outputs, both with key-spend and script-spend. BDK actually delegates
+most of the hard work to [rust-miniscript], and luckily most of the Taproot code was already implemented by the time I started working on it. I only had to patch a few little bugs here and there, and it ended up
+working flawlessly in the end.
+
+In this first part I will focus on the changes made to our dependencies, [rust-bitcoin] and [rust-miniscript]. In the second part I will talk about BDK itself.
+
+## Backstory
+
+On the evening of Thursday, November 11th I was attending our weekly [Satoshi Spritz] meetup in Milan. The activation of Taproot was right around the corner, and naturally that was the main discussion
+topic that night. The activation was forecasted for the early afternoon of Sunday, November 14th, a little less than 72h later.
+
+I began to wonder how hard it would be to patch BDK and add support for Taproot. I knew most of the work had already been done in our main dependencies, [rust-bitcoin] and [rust-miniscript], and so I decided
+to challenge myself: could I make it in time for the activation?
+
+The following day I started digging into the topic. It didn't help that up until that time I only had a rather "high level" idea of how Taproot worked, but luckily all the BIPs were very well written and
+straightforward to understand.
+
+By Friday night (or rather, early Saturday morning) [I had Taproot key-spend working][first-key-spend], which made me pretty optimistic even though the activation date was actually moving closer, now being forecasted for
+Sunday *morning*.
+
+After a few hours of sleep I went back to work and by early Saturday afternoon [I had Taproot key-spend working as well][first-script-spend]. This left me a few hours to coordinate with some friends and [generate a vanity address][vanity-addr]
```

Comment:
Did you mean "key-spend" here? or was this when you had the tap tree (or whatever the MAST stuff is called) working?